Sources/Sentry/SentryWatchdogTerminationTracker.m

```diff
@@ -97,6 +81,46 @@ - (void)start
 #endif // SENTRY_HAS_UIKIT
 }
 
+- (void)addBreadcrumbsToEvent:(SentryEvent *)event
+{
+    // Set to empty list so no breadcrumbs of the current scope are added
+    event.breadcrumbs = @[];
+
+    // Load the previous breadcrumbs from disk, which are already serialized
+    event.serializedBreadcrumbs = [self.fileManager readPreviousBreadcrumbs];
+    if (event.serializedBreadcrumbs.count > self.options.maxBreadcrumbs) {
+        event.serializedBreadcrumbs = [event.serializedBreadcrumbs
+            subarrayWithRange:NSMakeRange(
+                                  event.serializedBreadcrumbs.count - self.options.maxBreadcrumbs,
+                                  self.options.maxBreadcrumbs)];
+    }
+
+    NSDictionary *lastBreadcrumb = event.serializedBreadcrumbs.lastObject;
+    if (lastBreadcrumb && [lastBreadcrumb objectForKey:@"timestamp"]) {
+        NSString *timestampIso8601String = [lastBreadcrumb objectForKey:@"timestamp"];
+        event.timestamp = sentry_fromIso8601String(timestampIso8601String);
+    }
+}
+
+- (void)addContextToEvent:(SentryEvent *)event
+{
+    // Load the previous context from disk, or create an empty one if it doesn't exist
+    NSDictionary<NSString *, NSDictionary<NSString *, id> *> *previousContext =
+        [self.scopeContextStore readPreviousContextFromDisk];
+    NSMutableDictionary *context =
+        [[NSMutableDictionary alloc] initWithDictionary:previousContext ?: @{}];
```

Comment:
`l`: `readPreviousContextFromDisk` could return an empty dict instead of nil. Then we could simplify the logic here.

Reply:
Separate of concern. For the `readPreviousContextFromDisk` an empty context and a `nil` context is not the same.